### ISSUE TITLE
Dedicated profile page for student.

### ISSUE THREAD

**Issue by Ashrockzzz2003** [Ashrockzzz2003/placement_tracker_web#4]
> [!Note]
> Add the page in `/dashboard/student/profile`.

Some profile data is shown in the student's dashboard but the complete data is missing a dedicated page. The below data is what we get after login and will be accessible in dashboard page with `currentUser` from `secureLocalStorage`. Use it and build a profile page for the student. Add relevant links too in the page.

```json
{
    "message": "Student logged in!",
    "studentEmail": "cb.en.u4cse21001@cb.students.amrita.edu",
    "studentName": "Abhinav Ramakrishnan",
    "studentRollNo": "CB.EN.U4CSE21001",
    "studentId": 468,
    "studentSection": "A",
    "studentGender": "M",
    "studentBatch": "2025",
    "studentDept": "CSE",
    "isHigherStudies": "0",
    "isPlaced": "0",
    "CGPA": "9.12"
}
```

It needs the backend to be running locally too. Checkout `README.md` to find a link to backend code and visit that repo to further understand steps to run.

**Comment by suthekshan** [comment]
@Ashrockzzz2003 I would like to work on this.

**Comment by Ashrockzzz2003** [comment]
Assigned @suthekshan 
Deadline - `Jan 7, 2025`

**Comment by Ashrockzzz2003** [comment]
Please pull the latest version before making any changes!!
Sorry if you had already started working!

**Comment by suthekshan** [comment]
@Ashrockzzz2003 how is this design anna, should i add anything else?

![Screenshot 2025-01-04 231654](https://github.com/user-attachments/assets/ac7862ce-32b5-4856-85e1-6d01e2b37ec3)

**Comment by Ashrockzzz2003** [comment]
@TharunKumarrA

Can you add your suggestions please!

**Comment by TharunKumarrA** [comment]
Since there is no profile pic and all for this one coming from backend ... 
There is no proper clarity on the importance order (where eye goes first) ...
And overall it just blunt ...

So here are my suggestions to improve the design:
- Can have the name's font size bigger (but handle edge case for longer name) 
- next to the name in the right end have the edit button (no need for the edit profile text)
- Follow some hierarchical order when adding those elements - for example Dept section are related things, but you have them in 2 different columns. So try to have them together
- In here after name and rollno/email the main thing is whether he/she is placed or not ... that is yes or no value (so have it like a label/ tag next to the name or below it - red or green)
- One more suggestion is if you decide to go with like a red or green tag for the placement status ... we can have the opted for higher studies as yellow (instead of just plainly showing the text of what all is coming from backend try to use them as conditional vars to change a visually appealing element)

Hope these suggestions are helpful
@Ashrockzzz2003 @suthekshan

**Comment by TharunKumarrA** [comment]
Try to have like a visual hierarchy so its easy for anyone who see the profile page to know what's important - and get that first when viewing ... I think that's the key

**Comment by suthekshan** [comment]
Ok anna, I will work on these changes

**Comment by Ashrockzzz2003** [comment]
Please pull the latest version before making any changes!!
Major update in backend routes.
Sorry if you had already started working!

**Comment by suthekshan** [comment]
Updated the profile page anna. 
I have added the details in a hierarchy and related data is now grouped together
Is this fine or should I make any other changes anna?

@TharunKumarrA @Ashrockzzz2003 

![image](https://github.com/user-attachments/assets/7db9f691-7813-4908-857a-e20095d8cf99)


![image](https://github.com/user-attachments/assets/626db685-6f38-4b42-bf2f-add27857aa6c)
 Edge case if the name is longer

**Comment by TharunKumarrA** [comment]
Looks better @suthekshan 
But @Ashrockzzz2003 anna we won't have opted for higher studies if he is placed right?
That alone can you clarify anna

**Comment by Ashrockzzz2003** [comment]
More rounded corners?
Data part don't worry @TharunKumarrA. Backend will handle that!

Also can you try something creative @suthekshan.

**Comment by suthekshan** [comment]
![image](https://github.com/user-attachments/assets/ad98655e-fcfc-4336-874e-1dbe21cd0f94)
Is this fine anna? I added a background gradient, glass box, made the edit button blue, and reduced the rounded corners should I make any other changes?

@Ashrockzzz2003

**Comment by Ashrockzzz2003** [comment]
Check out the profile UI for manager and admin that were merged.
Follow a similar style and let's finish this

**Comment by suthekshan** [comment]
Yes, anna i will check and make the changes, I was waiting for your reply to proceed further.
